scripts/coreos-installer-service

```diff
@@ -44,17 +47,29 @@ copy_arg() {
     fi
 }
 
+# Config files
+have_config_file=
+for file in ${INSTALLER_CONFIG_DIR}/*; do
```

Comment:
I don't see any benefit to requiring a particular extension, and it adds a footgun.  We don't need to worry about RPM backup files, editor backup files, etc., because the files will be placed in the directory by Ignition, exist for one boot, and then be deleted.  The footgun is that files lacking the extension would be ignored, which could be unpleasant for users to debug.

Reply:
Don't feel strongly about this, but IMO the benefit is simply that it's more descriptive and enforces consistency.

> The footgun is that files lacking the extension would be ignored, which could be unpleasant for users to debug.

Hmm, I'm not sure how much of a concern that is. systemd does the same for many (all?) of its dropins directories like tmpfiles.d, sysusers.d, etc...

Reply:
I looked at a bunch of packages with `.d` directories in `/etc` on my system, and results were mixed:

### Require extension

- auditd
- dbus
- dhclient
- dnf
- dracut
- fwupd
- kmod
- ld.so (RPM package)
- NetworkManager
- ostree
- polkit
- sshd (RPM package)
- Xorg

### Do not require extension

- bash
- dconf
- dnsmasq (RPM package)
- grub
- kdump
- logrotate
- PAM
- sudo

It looks like there may be a trend in newer tools of requiring an extension, which makes sense for RPM packages that ship conffiles.  In this case I'm still not convinced that the extra debug overhead (for admins who may not be familiar with the convention) is worth it, though.  (And it's inconsistent with the `--config-file` option, which doesn't care about extensions.)

I'm going to go ahead and merge for now, and we can revisit in a followup if desired.